operators/src/processing/bandwise_expression/mod.rs

```diff
@@ -0,0 +1,438 @@
+use std::sync::Arc;
+
+use crate::engine::{
+    CanonicOperatorName, ExecutionContext, InitializedRasterOperator, InitializedSources, Operator,
+    OperatorName, QueryContext, RasterOperator, RasterQueryProcessor, RasterResultDescriptor,
+    ResultDescriptor, SingleRasterSource, TypedRasterQueryProcessor, WorkflowOperatorPath,
+};
+
+use crate::util::Result;
+use async_trait::async_trait;
+use futures::stream::BoxStream;
+use futures::{StreamExt, TryStreamExt};
+use geoengine_datatypes::primitives::RasterQueryRectangle;
+use geoengine_datatypes::raster::{
+    GridOrEmpty2D, MapElementsParallel, Pixel, RasterDataType, RasterTile2D,
+};
+use geoengine_expression::{
+    DataType, ExpressionAst, ExpressionParser, LinkedExpression, Parameter,
+};
+use serde::{Deserialize, Serialize};
+
+use super::expression::get_expression_dependencies;
+use super::RasterExpressionError;
+
+#[derive(Debug, Clone, PartialEq, Eq, Serialize, Deserialize)]
+#[serde(rename_all = "camelCase")]
+pub struct BandwiseExpressionParams {
+    pub expression: String,
+    pub output_type: RasterDataType,
+    pub map_no_data: bool,
+    // TODO: new unit for each band?
+}
+
+/// This `QueryProcessor` performs a unary expression on all bands of its input raster series.
+pub type BandwiseExpression = Operator<BandwiseExpressionParams, SingleRasterSource>;
+
+impl OperatorName for BandwiseExpression {
+    const TYPE_NAME: &'static str = "BandwiseExpression";
+}
+
+#[typetag::serde]
+#[async_trait]
+impl RasterOperator for BandwiseExpression {
+    async fn _initialize(
+        self: Box<Self>,
+        path: WorkflowOperatorPath,
+        context: &dyn ExecutionContext,
+    ) -> Result<Box<dyn InitializedRasterOperator>> {
+        let name = CanonicOperatorName::from(&self);
+
+        let source = self.sources.initialize_sources(path, context).await?.raster;
+
+        let in_descriptor = source.result_descriptor();
+
+        // TODO: ensure all bands have same measurement unit?
+
+        let result_descriptor = in_descriptor.map_data_type(|_| self.params.output_type);
+
+        let parameters = vec![Parameter::Number("x".into())];
+
+        let expression = ExpressionParser::new(&parameters, DataType::Number)
+            .map_err(RasterExpressionError::from)?
+            .parse(
+                "expression", // TODO: what is the name used for?
```

Comment:
what happens if we change it? Should we change it to be able to separate different kinds of exression?

Reply:
this seems to be the name of the rust function that is generated. I don't know why it matters. In the other raster expression it's either the bandname or just "expression".

what do you mean with "kinds of expressions"?